src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
+
+          // Clear socket data; we don't care what InstrumentApp sends
+          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
+          do
+            {
+              socket.Receive(buffer);
+            }
+          while (socket.Available != 0);
```

Comment:
I think you can only ever receive them once each, but we should double
check that.

On Thursday, April 28, 2016, Nathan D Acuff <notifications@github.com>
wrote:

> In src/Collector.cs
> <https://github.com/Instrumental/Instrumental.NET/pull/4#discussion_r61523718>
> :
>
> > +    {
> > +      // Is there any data available?
> > +      byte[] buffer = null;
> > +      while (socket.Poll(1, SelectMode.SelectRead))
> > +        {
> > +          // If no data is available then socket disconnected
> > +          if (socket.Available == 0)
> > +            return true;
> > +
> > +          // Clear socket data; we don't care what InstrumentApp sends
> > +          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
> > +          do
> > +            {
> > +              socket.Receive(buffer);
> > +            }
> > +          while (socket.Available != 0);
>
> it was possible (even likely) for the original ralph agent to get into a
> situation where it would send hello + authenticate over and over, as fast
> as possible, while also sending metrics.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Instrumental/Instrumental.NET/pull/4/files/71e433c010fcd20a21ab41cce81d43be4af75643#r61523718>
>